envoyauth/evaluation.go

```diff
@@ -30,46 +30,42 @@ type EvalContext interface {
 //Eval - Evaluates an input against a provided EvalContext and yields result
 func Eval(ctx context.Context, evalContext EvalContext, input ast.Value, result *EvalResult, opts ...func(*rego.Rego)) error {
 
-	err := storage.Txn(ctx, evalContext.Store(), storage.TransactionParams{}, func(txn storage.Transaction) error {
-		err := getRevision(ctx, evalContext.Store(), txn, result)
-		if err != nil {
-			return err
-		}
-
-		result.TxnID = txn.ID()
+	err := getRevision(ctx, evalContext.Store(), result.Txn, result)
```

Comment:
Since the `Eval()` function is exported now (instead of being hidden under `internal/`), someone could be calling. If someone is calling it, this change is going to break them because the function doesn't open a transaction anymore. I could see two approaches here:

1. Change the function signature so that at least callers break and can't compile
2. Make the change backwards compatible so that the code opens a txn if needed

I think that (2) wouldn't be too difficult with a bit of refactoring to avoid duplication, but I'd have to write the code myself to find out. Could you see if it's doable?

Reply:
Refactored the code a bit to make changes backwards compatible. LMKWYT.